cmd/test-server/networking.go

```diff
@@ -37,20 +37,20 @@ type Results struct {
 	AvgIcmpStat float64
 }
 
-// IcmpPinger sends ICMP pings and returns statistics
-func IcmpPinger(ip string) PingMsmt {
+// icmpPinger sends ICMP pings and returns statistics
+func icmpPinger(ip string) (PingMsmt, error) {
 	pinger, err := ping.NewPinger(ip)
 	if err != nil {
-		panic(err)
+		return PingMsmt{}, err
 	}
 	pinger.Count = icmpCount
 	pinger.Timeout = icmpTimeout
 	err = pinger.Run() // Blocks until finished.
 	if err != nil {
-		panic(err)
+		return PingMsmt{}, err
 	}
 	stat := pinger.Statistics()
 	icmp := PingMsmt{ip, stat.PacketsSent, stat.PacketsRecv, stat.PacketLoss, fmtTimeMs(stat.MinRtt),
```

Comment:
Nitpick: Maybe rename this, for clarity?
```suggestion
	pingMsmt := PingMsmt{ip, stat.PacketsSent, stat.PacketsRecv, stat.PacketLoss, fmtTimeMs(stat.MinRtt),
```

Reply:
Done in 3f2d7ef5826fd4ee72bcdf6b4a9cd4959227d56d